### PR TITLE
Add lower level Tab APIs

### PR DESCRIPTION
**Pull request by jquense**
not ready, but structurally sound. damn animations are brittle I hate them and I am tired of working on this right now.

also the way I hacked in a11y support with context, is terrifying, looking for alternatives

### CODE REVIEW

**Comment by taion** [comment]
This looks really cool BTW. Will take a closer look over the weekend.

**Comment by taion** [comment]
ref #1418

File: src/TabContainer.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    let ids = this._ids;
+    let old = ids[type].get(key);
+
+    if (old !== id) {
+      ids[type].set(key, id);
+      this._needsUpdate = true;
+    }
+
+    clearTimeout(this._timer);
+    this._timer = setTimeout(()=> {
```
Comment (taion) [inline src/TabContainer.js]: Why not put `ids` on `state`?

**Comment by taion** [comment]
- It's a little odd to have the tab-specific stuff in `<Nav>`. If nothing else, probably we don't want the weight in bundles for users who aren't using `<Tabs>`? Maybe we need `<TabsNav>` or something... ick.
- That a11y support... we could just not support automatically setting `aria-labelledby` if the user is using his own `id`s on the `<NavItem>`s.

**Comment by jquense** [comment]
yeah I felt that way about the tab stuff in Nav initially although as I did it,  it actually seemed like it is the right place. Nav is, in large part, there specifically for tabs (or similarly pills), it started to not feel weird the focus stuff was there. 

I am a bit ambivalent though to it, adding yet another Component also felt confusing. this seemed like the better tradeoff at the time. 

the a11y stuff is really ugly tho I am generally ok with that if it's not leaky, but I'm not sure it isn't :p

File: src/TabContainer.js (hunk 1 above)
Comment (jquense) [inline src/TabContainer.js]: it was an attempt to not over render since it would fire for each tab pane. no other reason tho

Comment (taion) [inline src/TabContainer.js]: Well, `setState` would coalesce all of those anyway, no?

**Comment by taion** [comment]
Hmm, on the `<Nav>`... I don't know... the navbar stuff is pretty lightweight, so it seems like it fits in nicely. The tab handling seems to have meaningful additional complexity.

The a11y stuff is a bit leaky, though, in that it won't work for server-side rendering, since the initial render won't pick up the custom `id`s. Do you use that feature? If not, I feel like the complexity and the SSR edge-casiness suggests that we might want to drop that bit.

**Comment by jquense** [comment]
closing to reopen against the rc branch

**Pull request closed without merging**